### PR TITLE
Link avatar image

### PR DESCRIPTION
**Pull request by Envl**
Things work now, but the with..... Files look quite counter-intuition to me.
Since they just attach some object to a component, there might be other clearer solutions.
I might look at that if there will be time after submitting the mid-project.

### CODE REVIEW

File: src/components/Account/Avatar.js [hunk 1]
```diff
@@ -26,7 +26,18 @@ const Avatar = props => {
 
   const postUploadTask = url => {
     setImgURL(url);
+
     //link the file to the user
+    props.firebase.auth.currentUser
+      .updateProfile({
+        photoURL: url,
+      })
+      .then(function() {
```
Comment (bt-s) [inline src/components/Account/Avatar.js:24]: Maybe write some console output here; now the then and catch of this Promise do nothing.

File: src/components/Account/Avatar.js [hunk 2]
```diff
@@ -77,11 +87,11 @@ const Avatar = props => {
 };
 
 Avatar.propTypes = {
-  avatarURL: PropTypes.string
+  avatarURL: PropTypes.string,
```
Comment (bt-s) [inline src/components/Account/Avatar.js:76]: @Envl it looks like your Prettier configuration has changed.
Throughout the code base, I have disabled trailing commas after the last item of a list or hashmap.

File: src/components/Authentication/SignUp.js [hunk 3]
```diff
@@ -57,11 +57,23 @@ const SignUpFormBase = props => {
     props.firebase
       .doCreateUserWithEmailAndPassword(form.email, form.passwordOne)
       .then(authUser => {
+        authUser.user
+          .updateProfile({
+            displayName: form.username,
+            photoURL:
+              'https://firebasestorage.googleapis.com/v0/b/bookio.appspot.com/o/images%2Frobot.png?alt=media&token=d0cc8a48-aaad-4e3b-9c08-3e341ed8165e',
+          })
+          .then(function() {
```
Comment (bt-s) [inline src/components/Authentication/SignUp.js]: Same here, please add some proper error handling.

File: src/components/Authentication/SignUp.js [hunk 4]
```diff
@@ -57,11 +57,23 @@ const SignUpFormBase = props => {
     props.firebase
       .doCreateUserWithEmailAndPassword(form.email, form.passwordOne)
       .then(authUser => {
+        authUser.user
+          .updateProfile({
+            displayName: form.username,
+            photoURL:
+              'https://firebasestorage.googleapis.com/v0/b/bookio.appspot.com/o/images%2Frobot.png?alt=media&token=d0cc8a48-aaad-4e3b-9c08-3e341ed8165e',
```
Comment (bt-s) [inline src/components/Authentication/SignUp.js]: Should this URL be hard-coded in here? If so, maybe define it as some sort of const or macro at the top of the file.

File: src/components/Firebase/firebase.js [hunk 5]
```diff
@@ -96,11 +96,12 @@ class Firebase {
 
             // merge auth and db user
             authUser = {
-              uid: authUser.uid,
-              email: authUser.email,
-              emailVerified: authUser.emailVerified,
-              providerData: authUser.providerData,
-              ...dbUser
+              // uid: authUser.uid,
```
Comment (bt-s) [inline src/components/Firebase/firebase.js:99]: It seems like these comments could be deleted.

File: src/components/Session/withAuthentication.js [hunk 6]
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
     constructor(props) {
       super(props);
 
-      this.props.onSetAuthUser(JSON.parse(localStorage.getItem('authUser')));
+      // this.props.onSetAuthUser(JSON.parse(localStorage.getItem('authUser')));
```
Comment (bt-s) [inline src/components/Session/withAuthentication.js]: It seems like this comment could be deleted.

File: src/components/Session/withAuthentication.js [hunk 7]
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
...
     }
 
     componentDidMount() {
       this.listener = this.props.firebase.onAuthUserListener(
         authUser => {
-          localStorage.setItem('authUser', JSON.stringify(authUser));
+          // localStorage.setItem('authUser', JSON.stringify(authUser));
```
Comment (bt-s) [inline src/components/Session/withAuthentication.js:13]: Same for this one.

File: src/components/Session/withAuthentication.js [hunk 8]
```diff
@@ -10,17 +10,17 @@ const withAuthentication = Component => {
...
           this.props.onSetAuthUser(authUser);
         },
         () => {
-          localStorage.removeItem('authUser');
+          // localStorage.removeItem('authUser');
```
Comment (bt-s) [inline src/components/Session/withAuthentication.js:17]: Same for this one.

**Review by bt-s — APPROVED** [review]
Nice catch!

The with... components are HOCs. This is quite idiomatic in React (have a look at withRouter from the react-router-dom package for example). I think that we should generally not change this, however, if you have an alternative that you would like to suggest I would be really interested in hearing it!

Please also have a look at the commits that I have pushed.

File: src/components/Authentication/SignUp.js (hunk 3 above)
Comment (Envl) [reply src/components/Authentication/SignUp.js]: Just according to the standard, we should catch the error, but there are no errors now. So I don't know what to catch.

File: src/components/Authentication/SignUp.js (hunk 4 above)
Comment (Envl) [reply src/components/Authentication/SignUp.js]: That's good. I just uploaded a default picture for all new users.

**Pull request merged**